cli/listener/src/main.rs

```diff
@@ -668,15 +703,19 @@ fn serve_request(
             Response::from_data(out).with_header(content_type).boxed()
         }
         Ok(Endpoint::IntervalPriceRequest(range_date)) => {
-            get_interval_price_request(db_connection, &range_date)
+            get_interval_price_request(db_connection, &range_date, now, current_slot)
         }
         Err(err) => get_error_response(err),
     };
     request.respond(response)
 }
 
 /// Spawn threads that run the http server.
-fn start_http_server(opts: &Opts, metrics_mutex: Arc<MetricsMutex>) -> Vec<JoinHandle<()>> {
+fn start_http_server(
+    opts: &Opts,
+    metrics_mutex: Arc<MetricsMutex>,
+    current_slot: Slot,
```

Comment:
This makes the current slot a constant that gets set at server startup and never updates!

But so let’s think about this a bit more ... on the one hand I like this approach of basing the `Expires` on just the current slot and the `IntervalPrices`, then we can have one uniform way of handling things. But on the other hand, I think it would be better to base it on the request:

* If you request with an absolute begin and end range, if you actually got the range you asked for, then that value will never change any more, even if the end happens to fall in the current epoch.
* If you request with `?days=30`, then it’s a moving target, and we do need to make it expire at the epoch boundary.

Maybe what we can do, instead of including the current slot, is just to pass in whether the request used a relative date range or not, and base the decision for how long to cache on that?

Reply:
>This makes the current slot a constant that gets set at server startup and never updates!

True, we need to include the clock in the snapshot then to use it, I suppose.
I think the approach is also good and wouldn't change much the end result. We can do it like you described.